### DISCUSSION TITLE
How to set up obsidian for editing content at Life Itself

### DISCUSSION THREAD

**Discussion by matthewmccarthy11** [life-itself/community#927] [General]
We use obsidian to [capture and organize content and data in markdown](https://github.com/orgs/life-itself/discussions/829).

This guide assumes you have downloaded obsidian and have a vault.

This guide is **only** about how to **configure obsidian in a few key ways.**

## Configure an assets folder

What: set up a dedicated assets folder in your obsidian vault.

Why: storing images and other assets in a specific folder keeps the vault tidier and is important for publishing.

### How

- Open Obsidian and go to Settings > Files & Links.
- Under Default location for new attachments, click ' In the folder specified below'. Then specify the folder you want in next option "Attachment folder path". Note: at Life Itself the folder is usually just called `assets`
- Click Save.

Obsidian will now store all new attachments in the assets folder.

## Configure links to be relative path in vault

What: setup obsidian so that internal links to files use relative paths.

Why: this works better for publishing and make things more portable generally.

### How

- Open Obsidian and go to Settings > Files & Links.
- Under Link format, select "Relative path to file".

Obsidian will now create relative links to files when you create new links or update existing links.

<img width="1154" alt="image" src="https://github.com/life-itself/community/assets/180658/a0618ffe-63b4-4991-9be3-43b2c4e5c918">

## Indent using 2 spaces

What: set your editor to indent with spaces of size 2

Why: consistency. See our style guide https://www.datopian.com/playbook/style-guide/markdown

### How

- Go to configuration
- Go to Editor
- Set `Indent using tabs` to off
- Set `Tab indent size` to 2 (smallest setting)

<img width="1089" alt="image" src="https://github.com/life-itself/community/assets/180658/55b56c54-4c3c-44c4-b440-5c90805285b3">

---

# Extras

## Obsidian vaults and github repos - how they relate and why we have a content folder (usually!)

What: when using obsidian to edit a markdown-based website managed in git(hub) ... you usually want to open  the content folder as the vault rather than the root folder of the whole repo ...

Why: the repo often contains a bunch of non-markdown content e.g. code for building the website, database files etc. You don't want that stuff in obsidian. Furthermore, if you have the repo root folder as the root of your obsidian vault markdown links will be wrong for when we publish the markdown ...

### How

Suppose your git repo layout is like this ...

```
README.md
.... # a bunch of other files and folders
content/
```

Then when opening a vault in obsidian choose the `content` folder inside the repo as  the folder rather than the repo folder itself.

## Templates in obsidian

What:

Why:

How: Link to the obsidian docs!

## Image links in frontmatter should use absolute path

What: When putting images in frontmatter, make sure to use the absolute path, not the relative path. eg. /assets/nameofpicture.png

Why: so that when we publish the site we can use the image path at other locations e.g. when creating cards in a listing (e.g. a list of blog posts) without breaking the url.

**Comment by catherinet1** [comment]
Very helpful to have this gathered so clearly and concisely in one place, thanks for putting this together @matthewmccarthy11 and @rufuspollock 🙏
